user-guide/Advanced_Functionality/DataMiner_Systems/BPA_tests/BPA_Check_Agent_Presence.md

```diff
@@ -0,0 +1,54 @@
+---
+uid: BPA_Check_Agent_Presence
+---
+
+# Check Agent Presence Test In NATS
+
+NATS serves as an inter-DMA message broker. This means that every DataMiner Agent must be able to reach every other DataMiner Agent in the cluster over NATS.
+
+This BPA test will scan the NATS info as seen by this Agent. It will verify if it finds a route to every Agent it is aware of, and it will perform a NATS client connection test to each Agent.
+
+Depending on the configuration of the system, the NATS info is gathered from SLNet (via SLCloud.xml) or via BrokerGateway. An extra check is performed for cluster consistency if DataMiner uses BrokerGateway.
+
+This BPA test is available on demand. You can [run it in System Center](xref:Running_BPA_tests) (on the *Agents > BPA* tab). From DataMiner 10.3.12 onwards, it is available by default.
+
+## Metadata
+
+- Name: Check Agent Presence In Nats
+- Description: Polls the NATS routes, checks if every Agent is represented, and performs a NATS client connection test
+- Author: Skyline Communications
+- Default Schedule: Every day
+
+## Results
+
+### Success
+
+- No issues detected.
+
+Every Agent's IP was found as a route in the NATS server, and the NATS client connection to every Agent succeeded. No inconsistencies were found when using BrokerGateway.
+
+### Error
+
+- Possible issues with NATS nodes or NATS client communication between DataMiners in the cluster. See Details.
```

Comment:
Is this the exact message shown by the BPA test? This because we usually avoid "DataMiners". Ideally, this should be "DataMiner Agents" instead.